service/internal/tenantclient/client.go

```diff
@@ -83,11 +82,10 @@ func (c *TenantClient) K8sClient(ctx context.Context, obj interface{}) (k8sclien
 		}
 
 		k8sClient, err = k8sclient.NewClients(c)
-		if tenant.IsAPINotAvailable(err) {
-			return nil, microerror.Mask(notAvailableError)
-		} else if err != nil {
-			return nil, microerror.Mask(err)
+		if err != nil {
+			return nil, microerror.Maskf(notAvailableError, err.Error())
```

Comment:
This here is the actual "fix". On any error we act accordingly. The error matching and graceful handler cancelation is already in place everywhere.